### PR TITLE
Customize OpenBAS

### PR DESCRIPTION
**Pull request by johanah29**
The user will be able to customize the colors of the platform and add some logos.

Related issue: https://github.com/OpenBAS-Platform/openbas/issues/507

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/rest/settings/PlatformSettingsApi.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+import static io.openbas.config.SessionHelper.currentUser;
+import static io.openbas.database.model.Setting.SETTING_KEYS.*;
+import static io.openbas.database.model.User.ROLE_ADMIN;
+import static io.openbas.helper.StreamHelper.fromIterable;
+import static java.util.Optional.ofNullable;
+
+
+@RestController
+public class PlatformSettingsApi extends RestBehavior {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/settings/PlatformSettingsApi.java:43]: There are some duplicate code from SettingsApi file
This two files (SettingsApi & PlatformSettingsApi) need to be merge

File: openbas-api/src/main/java/io/openbas/rest/settings/PlatformSettingsApi.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    Optional<Setting> optionalSetting = ofNullable(dbSettings.get(themeKey));
+    if (optionalSetting.isPresent()) {
+      Setting updateSetting = optionalSetting.get();
+      updateSetting.setValue(value);
+      return updateSetting;
+    }
+    return new Setting(themeKey, value);
+  }
+
+  @GetMapping("/api/platform/settings")
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/settings/PlatformSettingsApi.java:141]: It's better to have one endpoint to handle PlatformSettings.
And not both: /api/platform/settings & /api/settings

We need to finish refactoring on the front-end side (I know, I changed my mind :D )
And keep the /api/settings uri to be consistent with the other one: /api/settings/theme/*

File: openbas-api/src/main/java/io/openbas/rest/settings/PlatformSettingsApi.java [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+import io.openbas.config.OpenBASConfig;
+import io.openbas.config.OpenBASPrincipal;
+import io.openbas.database.model.Setting;
+import io.openbas.database.model.Theme;
+import io.openbas.database.repository.SettingRepository;
+import io.openbas.injects.opencti.config.OpenCTIConfig;
+import io.openbas.rest.helper.RestBehavior;
+import io.openbas.rest.settings.form.ThemeInput;
+import io.openbas.rest.settings.response.OAuthProvider;
+import io.openbas.rest.settings.response.PlatformSettings;
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/settings/PlatformSettingsApi.java:12]: There are some unused import in your files.
IntelliJ Idea can handle it for you wirth this shortcut -> Ctrl+Alt+O

File: openbas-model/src/main/java/io/openbas/database/model/Theme.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+      return false;
+    }
+    Base base = (Base) o;
+    return id.equals(base.getId());
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(id);
+  }
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/model/Theme.java:108]: You can use lombok to limit boilerplate

File: openbas-front/src/admin/components/settings/ThemeForm.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        variant="standard"
+        fullWidth
+        label={t('Background color')}
+        placeholder={t('Default')}
+        error={!!errors.background_color}
+        helperText={errors.background_color && errors.background_color?.message}
+        control={control}
+        name="background_color"
+      />
+      &nbsp;
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/settings/ThemeForm.tsx:59]: I am curious about this way to do spacing.
Why are you using that ?

File: yarn.lock [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (RomuDeuxfois) [inline yarn.lock:1]: This file should not be committed, the yarn.lock file is located under openbas-front folder

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 6 above)
Comment (RomuDeuxfois) [inline yarn.lock:1]: You need to remove this file

File: openbas-front/src/reducers/Referential.js [hunk 7]
```diff
@@ -28,6 +28,7 @@ export const entitiesInitializer = Immutable({
     tags: Immutable({}),
     documents: Immutable({}),
     parameters: Immutable({}),
```
Comment (RomuDeuxfois) [inline openbas-front/src/reducers/Referential.js:30]: Still used ?

File: openbas-front/.gitignore [hunk 8]
```diff
@@ -35,3 +35,4 @@ yarn-error.log*
 /playwright-report/
 /blob-report/
 /playwright/.cache/
+yarn.lock
```
Comment (RomuDeuxfois) [inline openbas-front/.gitignore:38]: Need to be remove

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: openbas-front/src/actions/Application.js [hunk 9]
```diff
@@ -3,18 +3,34 @@ import * as Constants from '../constants/ActionTypes';
 import * as schema from './Schema';
 import { getReferential, postReferential, putReferential, simpleCall } from '../utils/Action';
 
-export const fetchParameters = () => (dispatch) => {
-  return getReferential(schema.arrayOfParameters, '/api/settings')(dispatch);
-};
-
 export const updateParameters = (data) => (dispatch) => {
```
Comment (guillaumejparis) [inline openbas-front/src/actions/Application.js:6]: rename updatePlatformParameters everywhere needed to be coherent with the others changes

File: openbas-front/src/actions/Schema.js
Comment (guillaumejparis) [inline openbas-front/src/actions/Schema.js:1]: Do we need to keep this entity ? 

export const parameters = new schema.Entity(
  'parameters',
  {},
  { idAttribute: 'setting_key' },
);

File: openbas-front/src/actions/Application.js (hunk 9 above)
Comment (johanah29) [reply openbas-front/src/actions/Application.js]: Done

**Review by guillaumejparis — APPROVED** [review]

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**